### ISSUE TITLE
no-unresolved not working for require

### ISSUE THREAD

**Issue by mctep** [benmosher/eslint-plugin-import#90]
Hello!
Thank you for your job.

My file:

```js
require('./foo.js');
import './foo.js';

```

When executing
```bash
eslint --no-eslintrc --parser babel-eslint --plugin import --rule '{"import/no-unresolved":2}' test.js
```

I get only one error for import declaration:
```
~/test.js
  2:8  error  Unable to resolve path to module './foo.js'  import/no-unresolved

✖ 1 problem (1 error, 0 warnings)
```

**Comment by benmosher** [comment]
I'm not sure what you're expecting. The `no-unresolved` rule only lints `import` statement paths; it will not inspect `require` calls.

**Comment by benmosher** [comment]
Marking this `invalid` and closing for now, but if I've misunderstood, let me know.

**Comment by benmosher** [comment]
Actually, upon reflection, this seems like a good improvement on the rule, and it shouldn't be tough to add. 👍

**Comment by benmosher** [comment]
Agh... on second thought, `require` is a bit of a different animal, since it's a dynamic `CallExpression`... I think in the interest of simplicity, I'm going to re-mark this `invalid` and close. (at least for now... I may start doing more around CommonJS soon.)

**Comment by mctep** [comment]
Ok. I understand your reasons.

BTW. I has not found any plugin for resolving paths for `require` calls. So I can write it, but it will use a `resolver` similar to yours. And I think it will be pretty to add it feature in your module.

Maybe it will be good decision to add additional rule like `no-unresolved-require`.
So, please, look at this commit https://github.com/mctep/eslint-plugin-import/commit/acd4b4508d551f7f800fdd06e5c64ec01f3d1113

**Comment by mctep** [comment]
And support `ArrayExpression` for `require` arguments is not necessary, I think :smile:

**Comment by benmosher** [comment]
You make a strong case. Feels like it should be separate, though... Maybe as a setting?

I'll take another look. 😄

**Comment by benmosher** [comment]
Check out my PR. I adapted your alternate rule into a `commonjs` setting. 

Why were you supporting `require([...])`, though? for AMD? or something else? I'm not familiar with an array argument for CommonJS.

**Comment by mctep** [comment]
Oh. It's great! Thank you very much.

About an array of arguments. Yes it is using for AMD, and it should not be using for CommonJS. I realized it after commit only :-)

**Comment by benmosher** [comment]
Added support for AMD via `amd: true` option, too. 😄 Thanks for your persistence!

**Comment by mctep** [comment]
:beers:
